src/main/java/org/folio/circulation/domain/notice/schedule/DueDateScheduledNoticeService.java

```diff
@@ -97,17 +95,15 @@ private ScheduledNoticeConfig createScheduledNoticeConfig(NoticeConfiguration co
       .build();
   }
 
-  public CompletableFuture<Result<LoanAndRelatedRecords>> rescheduleDueDateNotices(
+  public Result<LoanAndRelatedRecords> rescheduleDueDateNotices(
     LoanAndRelatedRecords relatedRecords) {
-    return rescheduleDueDateNotices(relatedRecords.getLoan())
-      .thenApply(r -> r.map(relatedRecords::withLoan));
-  }
+    Loan loan = relatedRecords.getLoan();
 
-  private CompletableFuture<Result<Loan>> rescheduleDueDateNotices(Loan loan) {
-    if (loan.isClosed()) {
-      return completedFuture(succeeded(loan));
+    if (!loan.isClosed()) {
```

Comment:
Why has this condition been inverted?

Reply:
This change was made to schedule notices in background. Now the method always returns succeeded result and notices are scheduled asynchronously.

Reply:
If the method always succeeds and is synchronous, why return a future result?

Reply:
What happens if scheduling notices fails?

How would a library staff member know this has happened?

Reply:
Previously the method returned a future result, but now it returns just result.

Reply:
Why does it return a result when it always succeeds?

Reply:
Seems like library member doesn't now that this happend. Actually previously it workek in the same way. Looking at the code in master I found that the method return future responsible for deleting loan by id and than notice scheduling executes asynchronously and always returnes succeeded result.
Here is the code from the master branch:
https://github.com/folio-org/mod-circulation/blob/master/src/main/java/org/folio/circulation/domain/notice/schedule/ScheduledNoticeService.java#L110
https://github.com/folio-org/mod-circulation/blob/master/src/main/java/org/folio/circulation/domain/notice/schedule/ScheduledNoticeService.java#L40